includes/RequestWiki/RequestWikiRequestViewer.php

```diff
@@ -355,74 +450,230 @@ function ( array $formData, HTMLForm $form ) use ( $request ) {
-					'p',
-					[],
-					wfMessage( 'requestwiki-edit-success' )->plain()
-				),
-				'mw-notify-success'
-			)
+		if ( isset( $formData['submit-handle'] ) ) {
+			$this->wikiRequestManager->startQueryBuilder();
+
+			if ( isset( $formData['handle-visibility'] ) ) {
+				if ( $this->wikiRequestManager->getVisibility() !== (int)$formData['handle-visibility'] ) {
+					$this->wikiRequestManager->suppress(
+						user: $user,
+						level: $formData['handle-visibility'],
+						log: true
+					);
+				}
+			}
+
+			// Handle locking wiki request
+			if ( $this->wikiRequestManager->isLocked() !== (bool)$formData['handle-lock'] ) {
+				$this->wikiRequestManager->setLocked( (bool)$formData['handle-lock'] );
+				$this->wikiRequestManager->tryExecuteQueryBuilder();
+				if ( $formData['handle-lock'] ) {
+					$out->addHTML( Html::successBox(
+						$this->context->msg( 'createwiki-success-locked' )->escaped()
+					) );
+					return;
+				}
+
+				$out->addHTML( Html::successBox(
+					$this->context->msg( 'createwiki-success-unlocked' )->escaped()
+				) );
+				return;
+			}
+
+			/**
+			 * HANDLE STATUS UPDATES
+			 */
+
+			// Handle approve action
+			if ( $formData['handle-action'] === 'approve' ) {
+				// This will create the wiki
+				$this->wikiRequestManager->approve( $user, $formData['handle-comment'] );
+				$this->wikiRequestManager->tryExecuteQueryBuilder();
+				$out->addHTML( $this->getResponseMessageBox() );
+				return;
+			}
+
+			// Handle onhold action
+			if ( $formData['handle-action'] === 'onhold' ) {
+				$this->wikiRequestManager->onhold( $formData['handle-comment'], $user );
+				$this->wikiRequestManager->tryExecuteQueryBuilder();
+				$out->addHTML( $this->getResponseMessageBox() );
+				return;
+			}
+
+			// Handle moredetails action
+			if ( $formData['handle-action'] === 'moredetails' ) {
+				$this->wikiRequestManager->moredetails( $formData['handle-comment'], $user );
+				$this->wikiRequestManager->tryExecuteQueryBuilder();
+				$out->addHTML( $this->getResponseMessageBox() );
+				return;
+			}
+
+			// Handle decline action (the action we use if handle-action is none of the others)
+			$this->wikiRequestManager->decline( $formData['handle-comment'], $user );
+			$this->wikiRequestManager->tryExecuteQueryBuilder();
+			$out->addHTML( $this->getResponseMessageBox() );
+		}
+	}
+
+	public function isValidComment( ?string $comment, array $alldata ): bool|Message {
+		if ( isset( $alldata['submit-comment'] ) && ( !$comment || ctype_space( $comment ) ) ) {
+			return $this->context->msg( 'htmlform-required' );
+		}
+
+		return true;
+	}
+
+	public function isValidSubdomain( ?string $subdomain, array $alldata ): bool|Message {
+		if ( !isset( $alldata['submit-edit'] ) ) {
+			// If we aren't submitting an edit we don't want this to fail.
+			// For example, we don't want an invalid subdomain to block
+			// adding a comment.
+			return true;
+		}
+
+		if ( !$subdomain || ctype_space( $subdomain ) ) {
```

Comment:
Again, ehy this may be true, having just zero is actually not something we want so I feel it isn't really worth fixing. At least not here maybe in a follow-up if it is actually something someone wants.